src/client/client4.ts

```diff
@@ -1,6 +1,7 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import * as rudderAnalytics from 'rudder-sdk-js';
```

Comment:
Should we pull this into a `rudder.ts` file instead of as a "side export" of the client? (Then this file can just import that file so we maintain the "singleton" nature.)

Reply:
I thought about it but for these two lines, it seemed odd creating a file.

```
import * as rudderAnalytics from 'rudder-sdk-js';
export {rudderAnalytics};
```

Do you see more than this in a file?

Reply:
Hmm, Let me know if you see more than this but I think it makes to create a file as the import on web look good. 

`import rudderAnalytics from 'client/rudder';`

Reply:
Nope, I think that's exactly what I expected in the file. The benefit might be to add a comment explaining that this is effectively a singleton and not to import this resource yourself.

0/5 on the import path -- both your last example, and the current approach of importing `{rudderAnalytics}` from `client` as exported by the index seem fine.

Reply:
Done